packages/jest-emotion/src/index.js

```diff
@@ -56,43 +56,76 @@ type Options = {
   DOMElements?: boolean
 }
 
-function filterEmotionProps(props = {}) {
+function filterEmotionProps(props = {}, shouldKnowStyles = true) {
   const {
     css,
     __EMOTION_TYPE_PLEASE_DO_NOT_USE__,
     __EMOTION_LABEL_PLEASE_DO_NOT_USE__,
     ...rest
   } = props
 
-  rest.css = 'unknown styles'
+  if (shouldKnowStyles) {
+    rest.css = 'unknown styles'
+  }
 
   return rest
 }
 
+function isShallowEnzymeElement(element, classNames) {
+  let renderedClassNames = (element.children || [])
+    .map(({ props = {} }) => props.className)
+    .filter(Boolean)
+    .join(' ')
+  return !classNames.some(className =>
+    new RegExp(className).test(renderedClassNames)
+  )
+}
+
 export function createSerializer({
   classNameReplacer,
   DOMElements = true
 }: Options = {}) {
   let cache = new WeakSet()
   function print(val: *, printer: Function) {
-    if (isEmotionCssPropEnzymeElement(val)) {
-      return val.children.map(printer).join('\n')
-    }
+    let elements = getStyleElements()
+    let keys = getKeys(elements)
     if (isEmotionCssPropElementType(val)) {
       return printer({
         ...val,
         props: filterEmotionProps(val.props),
         type: val.props.__EMOTION_TYPE_PLEASE_DO_NOT_USE__
       })
     }
+    if (isEmotionCssPropEnzymeElement(val)) {
```

Comment:
could you explain why things weren't working before and how they got fixed now?

Reply:
When the emotion element is shallow rendered, the actual element that would have the className prop is not rendered. The idea here is to swap the emotion element for an element of the underlying type (from __EMOTION_TYPE_PLEASE_DO_NOT_USE__) with the className prop added.